src/gsy_e/models/market/future.py

```diff
@@ -54,9 +88,46 @@ def __init__(self, bc: Optional[NonBlockchainInterface] = None,
                          readonly=readonly, grid_fee_type=grid_fee_type,
                          grid_fees=grid_fees, name=name, in_sim_duration=True)
 
-        self.slot_bid_mapping: Dict[DateTime, List[Bid]] = {}
-        self.slot_offer_mapping: Dict[DateTime, List[Offer]] = {}
-        self.slot_trade_mapping: Dict[DateTime, List[Trade]] = {}
+        self._offers = FutureOrders()
+        self._bids = FutureOrders()
+
+    @property
+    def offers(self) -> FutureOrders:
+        """Return the {offer_id: offer} mapping."""
+        return self._offers
+
+    @offers.setter
+    def offers(self, orders) -> None:
+        """Wrap the setter of _orders in order to build a FutureOrders object."""
+        self._offers = FutureOrders(orders)
```

Comment:
Each time we try to reassign the offers dict `market.offers = {"id1": offer1}`, this setter will get rid of the typical dictionary and will use our FutureOrders dictionary